File: boot-pinyougou-sellergoods-interface/src/main/java/com/xt/pinyougou/service/ItemCatService.java
package com.xt.pinyougou.service;

import com.baomidou.mybatisplus.extension.service.IService;
import com.xt.pinyougou.pojo.ItemCat;

import java.util.List;

/**
 * <p>
 * 商品类目 服务类
 * </p>
 *
 * @author xt
 * @since 2019-11-27
 */
public interface ItemCatService extends IService<ItemCat> {

    /**
     * 根据上级ID返回列表
     * @param parentId : 父 ID
     * @return
     */
    List<ItemCat> listByParentId(Long parentId);

    Boolean deleteBatch(List<Long> asList);
}

File: boot-pinyougou-shop-web/src/main/resources/static/service/itemCatService.js
// 品牌服务
app.service('itemCatService', function ($http) {

    // 根据上级ID查询子类列表
    this.listByParentId = function (parentId) {
        return $http.get('/itemCat/list/' + parentId);
    };

    // 详情
    this.findOne = function (id) {
        return $http.get('/itemCat/' + id);
    };

    // 全部列表
    this.findAll = function() {
        return $http.get('/itemCat/list');
    }

});